src/cft.ts

```diff
@@ -4,22 +4,25 @@ import { selectTestEnvironment } from "./environments/selectTestEnvironments";
 import { getSpyFactory } from "./spies/selectSpyFactory";
 import { MethodCall, MethodSpy } from "./spies/spyTypes";
 import { CftRequest } from "./types";
+import { setDefaults } from "./defaults";
 
-export const cft = (request: CftRequest = {}) => {
+export const cft = (_request: Partial<CftRequest>) => {
+    const request = setDefaults(_request);
     const spyFactory = getSpyFactory(request);
     const testEnvironment = selectTestEnvironment(request);
     const methodSpies: { [i: string]: MethodSpy } = {};
+    const relevantMethodNames = consoleMethodNames.filter((name) => !!request.console[name as keyof typeof request.console]);
```

Comment:
Ha, this. I'll take a look and see if I can wrestle TypeScript into not needing that `as`...

Reply:
Ooh! We need to mark `: CftRequest` on `setDefaults` because the type of `request.console` is being inferred to be `ConsoleObject | {}`, rather than `ConsoleObject`. As a result the built-in "it's a keyof the type, you can use it as an index" check being done for the `[name]` lookup isn't triggering.